File: python/2221.py
z = int(input())
i = 0

while i < z:
    bonus = int(input())
    a1, d1, l1 = map(int, input().split())
    a2, d2, l2 = map(int, input().split())
    
    golpe1 = (a1+d1) / 2 + bonus if l1%2 == 0 else (a1+d1) / 2
    golpe2 = (a2+d2) / 2 + bonus if l2%2 == 0 else (a2+d2) / 2
    
    if golpe1 > golpe2:
        print("Dabriel")
    elif golpe1 < golpe2:
        print("Guarte")
    else:
        print("Empate")
    i += 1

File: dart/2221.dart
import 'dart:io';

void main() {
  int n = int.parse(stdin.readLineSync()!);

  for (int i = 0; i < n; i++) {
    int bonus = int.parse(stdin.readLineSync()!);
    List<String> dabriel = stdin.readLineSync()!.split(" ");
    List<String> guarte = stdin.readLineSync()!.split(" ");

    double vd = (int.parse(dabriel[0]) + int.parse(dabriel[1])) / 2;
    double vg = (int.parse(guarte[0]) + int.parse(guarte[1])) / 2;

    if (int.parse(dabriel[2]) % 2 == 0) {
      vd += bonus;
    }
    if (int.parse(guarte[2]) % 2 == 0) {
      vg += bonus;
    }

    if (vd > vg) {
      print("Dabriel");
    } else if (vd < vg) {
      print("Guarte");
    } else {
      print("Empate");
    }
  }
}
